platform-operator/controllers/verrazzano/component/grafana/grafana_component.go

```
@@ -234,3 +235,18 @@ func (g grafanaComponent) ValidateUpdateV1Beta1(old *installv1beta1.Verrazzano,
 func (g grafanaComponent) Reconcile(ctx spi.ComponentContext) error {
 	return nil
 }
+
+// checkExistingGrafana checks if Grafana is already installed
+// OLCNE Istio module may have Grafana installed in istio-system namespace
+func checkExistingGrafana(vz runtime.Object) error {
+	if !vzconfig.IsGrafanaEnabled(vz) {
+		return nil
+	}
+	if err := k8sutil.ErrorIfDeploymentExists(constants.IstioSystemNamespace, ComponentName); err != nil {
```

shouldn't this be the verrazzano-system namespace, not istio-system?

---

Also, in my VZ install, the Grafana deployment is called vmi-system-grafana. There's a constant in the grafana package already called "grafanaDeployment" in grafana.go that you can use.

---

OLCNE install grafana and prometheus in istio-system namespace. Though that does not cause collision, we still want to report the error of existing grafana when installing.

---

yeah I understood that later - hence my suggestion of renaming that checkExisting function so that it is obvious that it's the CNE one it's looking for, not just any Grafana install.

---

changed to checkExistingCNEGrafana